src/condor/htcondor_helper.py

```diff
@@ -2,174 +2,332 @@
 import time
 import os
 from typing import Any
-from condor.utils import setup_logger
-
-# Get the centralized logger instance
-logger = setup_logger()
 
 class HTCondorHelper:
-    def __init__(self, config: 'dict[str, Any]'):
-        """Initialize HTCondor helper with configuration"""
+    def __init__(self, config, token_dir: str):
```

Comment:
is there a reason why token_dir cannot be a field of 'config'? and have one fewer parameter here. also I would like it if parameters have type annotations (I assume 'config' should be of type dict[str, Any] or Config)